### PR TITLE
(FACT-2691) Add kvm fact on linux

### PR DESCRIPTION
**Pull request by IrimieBogdan**
Added `kvm` resolver for Linux platforms.

### CODE REVIEW

File: spec/facter/facts/linux/hypervisors/kvm_spec.rb [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+        allow(Facter::Resolvers::Linux::DmiBios).to receive(:resolve).with(:product_name).and_return('Parallels')
+      end
+
+      it 'calls  Facter::Resolvers::Linux::DmiBios' do
+        fact.call_the_resolver
+
+        expect(Facter::Resolvers::Linux::DmiBios).to have_received(:resolve).with(:product_name)
+      end
+
+      it 'have nil value' do
```
Comment (Filipovici-Andrei) [inline spec/facter/facts/linux/hypervisors/kvm_spec.rb:35]: has a nil value, this applies also to the other similar tests.

File: spec/facter/facts/linux/hypervisors/kvm_spec.rb [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+    context 'when VM is provided by OpenStack with KVM hypervisor' do
+      before do
+        allow(Facter::Resolvers::Linux::DmiBios).to receive(:resolve).with(:product_name).and_return('KVM')
+        allow(Facter::Resolvers::Linux::DmiBios).to receive(:resolve).with(:bios_vendor).and_return('OpenStack')
+        allow(Facter::Resolvers::Linux::DmiBios).to receive(:resolve).with(:sys_vendor).and_return('OpenStack')
+        allow(Facter::Resolvers::VirtWhat).to receive(:resolve).with(:vm).and_return('unknown')
+        allow(Facter::Resolvers::Lspci).to receive(:resolve).with(:vm).and_return('kvm')
+      end
+
+      it 'returns google cloud' do
```
Comment (Filipovici-Andrei) [inline spec/facter/facts/linux/hypervisors/kvm_spec.rb:121]: it 'returns OpenStack' do

File: spec/facter/facts/linux/hypervisors/kvm_spec.rb (hunk 1 above)
Comment (IrimieBogdan) [reply spec/facter/facts/linux/hypervisors/kvm_spec.rb]: changed

File: spec/facter/facts/linux/hypervisors/kvm_spec.rb (hunk 2 above)
Comment (IrimieBogdan) [reply spec/facter/facts/linux/hypervisors/kvm_spec.rb]: changed

**Pull request merged**